### PR TITLE
Added speed test.

### PR DESCRIPTION
**Pull request by gerring**
https://github.com/eclipse/january/issues/420

First run of tests will fail until @Ignore is added to test. This test is for future fixes or for developers to check changes in basic maths speed vs. a naive loop.

### CODE REVIEW

File: org.eclipse.january.asserts/.classpath [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
Comment (PeterC-DLS) [inline org.eclipse.january.asserts/.classpath]: Keep Java 8 changes to a separate PR

File: org.eclipse.january.test/.classpath [hunk 2]
```diff
@@ -1,7 +1,25 @@
...
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7"/>
-	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
-	<classpathentry kind="src" path="src"/>
-	<classpathentry kind="output" path="bin"/>
-</classpath>
+  <classpathentry kind="src" path="src" including="**/*.java"/>
+  <classpathentry kind="output" path="target/classes"/>
+  <classpathentry kind="var" path="M2_REPO/javax/measure/unit-api/1.0/unit-api-1.0.jar"/>
+  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+  <classpathentry kind="lib" path="C:/Users/matt/.m2/repository/p2/osgi/bundle/javax.measure.unit-api/1.0.0.v20170818-1538/javax.measure.unit-api-1.0.0.v20170818-1538.jar"/>
```
Comment (PeterC-DLS) [inline org.eclipse.january.test/.classpath]: Remove all your machine specific settings

File: org.eclipse.january.test/src/org/eclipse/january/dataset/SpeedTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
+package org.eclipse.january.dataset;
```
Comment (PeterC-DLS) [inline org.eclipse.january.test/src/org/eclipse/january/dataset/SpeedTest.java]: Place in a separate fragment and use JMH

File: org.eclipse.january/src/org/eclipse/january/dataset/LazyDatasetBase.java [hunk 4]
```diff
@@ -996,11 +996,11 @@ protected ILazyDataset createFromSerializable(Serializable blob, boolean keepLaz
 				if (is != 1 && is != getElementsPerItem()) {
 					throw new IllegalArgumentException("Dataset has incompatible number of elements with this dataset");
 				}
-				d = ed.cast(is == 1 ? DoubleDataset.class: CompoundDoubleDataset.class);
+				d = ed.cast((Class<Dataset>)(is == 1 ? DoubleDataset.class: CompoundDoubleDataset.class));
```
Comment (PeterC-DLS) [inline org.eclipse.january/src/org/eclipse/january/dataset/LazyDatasetBase.java]: Unrelated changes

Comment (gerring) [reply org.eclipse.january/src/org/eclipse/january/dataset/LazyDatasetBase.java]: This is part of Java 8 changes

File: org.eclipse.january.asserts/.classpath (hunk 1 above)
Comment (gerring) [reply org.eclipse.january.asserts/.classpath]: Ok
